site/viamguides/src/ncnn-edge-ai/ncnn-edge-ai.md

```diff
@@ -0,0 +1,246 @@
+   ```bash
+   sudo apt update
+   sudo apt upgrade -y
+   ```
+
+<!-- ------------------------ -->
+## Configure your machine
+Duration: 5
+
+### Connect your webcam
+
+1. Connect the webcam to any USB port on your Pi.
+
+### Configure your machine in Viam
+
+1. In [the Viam app](https://app.viam.com/fleet/dashboard) under the **LOCATIONS** tab, create a machine by typing in a name like "ncnn-edge" and clicking **Add machine**.
+   ![create new machine in Viam app](./assets/create-machine.png)
+1. Click **View setup instructions**.
+   ![new, unprovisioned machine in Viam app](./assets/new-machine.png)
+1. To install `viam-server` on the Raspberry Pi device that you want to use, select the `Linux / Aarch64` platform for the Raspberry Pi, and leave your installation method as [`viam-agent`](https://docs.viam.com/how-tos/provision-setup/#install-viam-agent).
+   ![machine setup instructions in Viam app](./assets/machine-setup.png)
+1. Use the `viam-agent` to download and install `viam-server` on your Raspberry Pi. Follow the instructions to run the command provided in the setup instructions from the SSH prompt of your Raspberry Pi.
+   ![viam agent setup script in SSH session](./assets/viam-agent-setup.png)
+1. The setup page will indicate when the machine is successfully connected.
+   ![machine connected successfully](./assets/machine-connected.png)
+
+<!-- ------------------------ -->
+## Configure your webcam
+Duration: 5
+
+1. In [the Viam app](https://app.viam.com/fleet/locations) under the **CONFIGURE** tab, click the **+** icon in the left-hand menu and select **Component or service**.
+   ![select component or service](./assets/add-component.png)
+1. Select `camera`, and find the `webcam` module. This adds the module for working with a USB webcam. Leave the default name `camera-1` for now.
+   ![add camera component](./assets/name-camera.png)
+1. Notice adding this component adds the webcam hardware component called `camera-1`. From the **Configure** section of the panel, switch to the JSON configuration mode using the `{}` icon in the top-right:
+   ![new camera component in configuration](./assets/new-camera-component.png)
+   ![switch configuration mode](./assets/camera-advanced-config.png)
+1. In this mode, configure your component with the following JSON in the **CONFIGURE** field. This will tell the component to look for the default camera connected to the device, which will be the USB webcam.
+   ```json
+   {
+       "video_path": ""
+   }
+   ```
+   ![set the blank video path](./assets/camera-component-config.png)
+1. Click **Save** in the top right. This may take a moment to apply your configuration changes.
+
+   > aside positive
+   > If multiple webcams are connected to your Pi, you'll need to specify the correct video path. To list available camera devices, you can use the [`rand:find-webcams:webcam-discovery`](https://app.viam.com/module/rand/find-webcams)
```

Comment:
Since the screenshot shows this toast 2 steps prior, may be helpful to include a screenshot of the toast in this callout here.